### DISCUSSION TITLE
Can the backup location be within Google Cloud Platform?

### DISCUSSION THREAD

**Discussion by bhavekost** [GAM-team/got-your-back#367] [Q&A]
Hello - 

Is there a method or syntax to use the `--local-folder` for a target within our GCP instance?

We backup our domain's email/drives quarterly into GCP thru the Google Admin console.   Currently, with over 1,000 users, we must contact Google support to kickoff the domain backup, and it takes about a week to run, and then copy TB's worth of data into our GCP instance.  

The goal we're searching for is to backup offboarded users accounts for historical retention.

We're investigating doing specific & targeted users Gmail backups with GYB, but rather than make it a multi-step process (GYB locally, then copy up to GCP), we would like to perform the GYB directly into our GCP storage area by username.  Additionally, I would assume that this is much fast since it would all be within the Google infrastructure and no upload/download might be needed.

Thank you in advance,
Brett

**Comment by jay0lee** [comment]
No. There is no support planned for GCS. you'd need to use the two step approach.

**Reply by bhavekost** [reply]
Thanks!

**Comment by nicktrejo** [comment]
Hi, @bhavekost 
Let me share with you our approach. Firstly, we are using extracting data from Vault and not the users Gmail mailbox (as GYB does so far).

Our process is the following (not fully automated yet):

1. Python script that using Vault APIs generates the Vault exports for each offboarding user.
2. Next, we run a python script that generates a bash file with the commands to organise and download all the files (info is extracted from he COMPLETED exports). This bash file runs on one of our Linux machines and generates a folder for each user and downloads their files
3. Finally we upload these folders to the Shared Drive, using the user interface. We delete the already uploaded users from the local machine

I uploaded the scripts in this repo: https://github.com/nicktrejo/google-vault-backup

Hope you find this usefull and feel free to give me your feedback
